### PR TITLE
Fix GeneratorsError for creating FractionField element with domain of PolynomialRing or FractionField

### LINKED ISSUE #19581: GeneratorsError raised when creating element of fraction field of polynomial ring
I see this construction is not possible
```python3
>>> from sympy import *
>>> a, b, x = symbols('a b x')
>>> domain = ZZ[a, b][x].get_field()
>>> domain.field(a+b)
sympy.polys.polyerrors.GeneratorsError: unable to drop generators
```
while it can be constructed from an element of `ZZ[a+b][x]`
```python
>>> domain.field(ZZ[a, b][x](a + b))
a + b
```

The same error raises for an element of `ZZ[a+b]`
```python
>>> domain.field(ZZ[a, b](a + b))
sympy.polys.polyerrors.GeneratorsError: unable to drop generators
```

So this can be relevant

### PR DESCRIPTION
**Pull request by sylee957**
#### References to other Issues or PRs

Fixes #19581

#### Brief description of what is fixed or changed

Fraction field can have domain of fraction field or polynomial ring 
and sometimes the coefficient itself can be the element from its domain.

This fixes two issues for constructing
`ZZ.frac_field(a, b).frac_field(x)(a+b)`
`ZZ.poly_ring(a, b).frac_field(x)(a+b)`

#### Release Notes

- polys
  - Fixed `GeneratorsError` for creating some elements of `FractionField` when its ground domain is `FractionField` or `PolynomialRing`.

### CODE REVIEW

**Comment by oscarbenjamin** [comment]
Looks good to me

File: sympy/polys/tests/test_fields.py [hunk 1]
```diff
@@ -137,6 +137,22 @@ def test_FracElement_from_expr():
         FracElement)
 
 
+def test_FracField_nested():
+    a, b, x = symbols('a b x')
+    F1 = ZZ.frac_field(a, b)
+    F2 = F1.frac_field(x)
+    numer = F2(a + b).numer
+    numer == F1.poly_ring(x)(a + b)
```
Comment (jksuom) [inline sympy/polys/tests/test_fields.py:145]: Should there be `assert`s?

Comment (sylee957) [reply sympy/polys/tests/test_fields.py:145]: Okay, I'll fix them

File: sympy/polys/fields.py [hunk 2]
```diff
@@ -190,11 +190,17 @@ def field_new(self, element):
...
+                return self.ground_new(element)
             else:
                 raise NotImplementedError("conversion")
         elif isinstance(element, PolyElement):
             denom, numer = element.clear_denoms()
-            numer = numer.set_ring(self.ring)
+            if isinstance(self.domain, PolynomialRing) and \
+                numer.ring == self.domain.ring:
+                numer = self.ring.ground_new(numer)
+            else:
```
Comment (jksuom) [inline sympy/polys/fields.py:202]: What could be done if `self.domain` is a `FractionField`?

Comment (sylee957) [reply sympy/polys/fields.py:211]: I've fixed them

**Comment by jksuom** [comment]
Thanks, looks good.

**Pull request merged**